### DISCUSSION TITLE
Widget provider state hidden not working [Evercam]

### DISCUSSION THREAD

**Discussion by salah-walid** [iTwin/itwinjs-core#2892] [Q&A]
At [Evercam](https://evercam.io/), we're trying to hide some custom widgets on certain conditions, but when setting it's state to WidgetState.Hidden like the code bellow doesn't do the trick:

```
public provideWidgets(_stageId: string, _stageUsage: string, location: StagePanelLocation, _section?: StagePanelSection): ReadonlyArray<AbstractWidgetProps> {
    const widgets: AbstractWidgetProps[] = [];
    if (location === StagePanelLocation.Left) {
      widgets.push(
        {
          id: "WidgetId",
          label: "Widget",
          defaultState: WidgetState.Hidden,
          // eslint-disable-next-line react/display-name
          getWidgetContent: () => <Widget />,
        }
      );
    }
    return widgets;
  }
```

is WidgetState.Hidden supposed to have another behaviour than to hide the widget ?

**Comment by bsteinbk** [comment]
This [PR](https://github.com/iTwin/itwinjs-core/pull/2643) fixes that issue for the upcoming 3.0 release, but it was not backported to any 2.x release.

**Reply by kmurari103** [reply]
Is it possible to hide the built in widgets (tree & properties) at the time app starts? 
And then make this widgets visible/invisible based on a button click?

**Reply by aruniverse** [reply]
This is not possible at this time with the default widgets. You'd have to hide them using the defaultUiConfig prop on the viewer, then update the widgets yourself, and write a UiProvider for them, and then add them to the Viewer.
